src/Raven.Server/Documents/Queries/Results/IQueryResultRetriever.cs

```diff
@@ -1,13 +1,63 @@
-﻿using Lucene.Net.Search;
+﻿using Corax;
+using Lucene.Net.Search;
 using Lucene.Net.Store;
-using Raven.Server.Documents.Indexes.Static.Spatial;
 
 namespace Raven.Server.Documents.Queries.Results
 {
     public interface IQueryResultRetriever
     {
-        Document Get(Lucene.Net.Documents.Document input, ScoreDoc lucene, IState state);
+        Document Get(ref RetrieverInput retrieverInput);
 
-        bool TryGetKey(Lucene.Net.Documents.Document document, IState state, out string key);
+        bool TryGetKey(ref RetrieverInput retrieverInput, out string key);
+    }
+
+    public ref struct RetrieverInput
+    {
+        public IndexEntryReader CoraxEntry
+        {
+            get;
+            set;
+        }
+
+        public IState State
+        {
+            get; 
+            set;
+        }
+        public Lucene.Net.Documents.Document LuceneDocument 
+        { 
+            get; 
+            set;
+        }
+
+        public string DocumentId
+        {
+            get; 
+            set;
```

Comment:
No need for properties, can be fields only, this is on the server side, after all

Reply:
Done.